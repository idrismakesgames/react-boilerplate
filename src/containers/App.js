import React, { Component } from 'react';
import '../global-styles/global.scss';
import classes from './App.module.scss';

class App extends Component {
  state = {
    person: 'testing',
  };

  testFunct = () => 'Add this';

  render() {
    return (
      <div className={classes.app}>
        {this.state.person} Sewhatatart! {this.testFunct()}
      </div>
    );
  }
}

export default App;
