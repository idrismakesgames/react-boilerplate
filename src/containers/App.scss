.app {
  width: 100px;
  height: 100px;
  background: blue;
  color: yellow;

  .class1 {
    .class2 {
      .class3 {
        background: black;

        .class4 {
          color: white;
        }
      }
    }
  }
}
