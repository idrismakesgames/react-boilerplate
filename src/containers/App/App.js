import React, { Component } from 'react';
import SearchForm from '../SearchForm/SearchForm';
import Header from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import '../../global-styles/global.scss';
import classes from './App.module.scss';

class App extends Component {
  state = {
    person: 'testing',
  };

  testFunct = () => 'Search Below';

  render() {
    return (
      <div className={classes.app} person={this.state.person}>
        <Header name="header" />
        {this.testFunct()}
        <SearchForm name="search" />
        <Footer name="footer" />
      </div>
    );
  }
}

export default App;
