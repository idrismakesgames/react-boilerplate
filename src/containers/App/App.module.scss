@import "../../global-styles/global.scss";

.app {
  background-color: #fff;
  width: 50%;
  margin: 20px auto;
}
