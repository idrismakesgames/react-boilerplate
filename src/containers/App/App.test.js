import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const app = shallow(<App />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  describe('Header ', () => {
    it('is contained within the `App` component, with a name prop', () => {
      expect(app.find('Header').exists()).toEqual(true);
      expect(app.find('Header')).toHaveProp('name', 'header');
    });
  });

  describe('Search Form ', () => {
    it('is contained within the `App` component, with a name prop', () => {
      expect(app.find('SearchForm').exists()).toEqual(true);
    });
  });

  describe('Footer ', () => {
    it('is contained within the `App` component, with a name prop', () => {
      expect(app.find('Footer').exists()).toEqual(true);
      expect(app.find('Footer')).toHaveProp('name', 'footer');
    });
  });
});
