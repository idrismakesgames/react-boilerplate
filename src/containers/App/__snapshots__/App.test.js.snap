// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="app"
  person="testing"
>
  <Header
    name="header"
  />
  Search Below
  <SearchForm
    name="search"
  />
  <Footer
    name="footer"
  />
</div>
`;
